sdk/src/wallet/operations/output_consolidation.rs

```diff
@@ -70,6 +73,67 @@ where
     crate::wallet::Error: From<S::Error>,
     crate::client::Error: From<S::Error>,
 {
+    /// Consolidates basic outputs from an account by sending them to a provided address or to an own address again if
+    /// the output amount is >= the output_threshold. When `force` is set to `true`, the threshold is ignored. Only
+    /// consolidates the amount of outputs that fit into a single transaction.
+    pub async fn consolidate_outputs(&self, params: ConsolidationParams) -> Result<TransactionWithMetadata> {
+        let prepared_transaction = self.prepare_consolidate_outputs(params).await?;
+        let consolidation_tx = self
+            .sign_and_submit_transaction(prepared_transaction, None, None)
+            .await?;
+
+        log::debug!(
+            "[OUTPUT_CONSOLIDATION] consolidation transaction created: block_id: {:?} tx_id: {:?}",
+            consolidation_tx.block_id,
+            consolidation_tx.transaction_id
+        );
+
+        Ok(consolidation_tx)
+    }
+
+    /// Prepares the transaction for [Wallet::consolidate_outputs()].
+    pub async fn prepare_consolidate_outputs(&self, params: ConsolidationParams) -> Result<PreparedTransactionData> {
+        log::debug!("[OUTPUT_CONSOLIDATION] prepare consolidating outputs if needed");
+        let wallet_address = self.data().await.address.clone();
+
+        let outputs_to_consolidate = self.get_outputs_to_consolidate(&params).await?;
+
+        let mut custom_inputs = Vec::new();
+        let mut commitment_context_input_required = false;
+        for output_data in outputs_to_consolidate {
+            custom_inputs.push(output_data.output_id);
+
+            if output_data.output.unlock_conditions().map_or(false, |uc| {
+                uc.expiration().is_some() || uc.storage_deposit_return().is_some()
+            }) {
+                commitment_context_input_required = true;
```

Comment:
That could also be added to build_transaction.rs

Reply:
Done